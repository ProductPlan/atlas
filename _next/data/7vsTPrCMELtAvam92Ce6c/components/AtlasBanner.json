{"pageProps":{"component":"AtlasBanner"},"__N_SSG":true}