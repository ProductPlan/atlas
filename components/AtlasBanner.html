<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width"/><title>ProductPlan Atlas</title><meta name="next-head-count" content="3"/><link rel="preconnect" href="https://fonts.googleapis.com"/><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="true"/><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.2/css/all.min.css"/><link rel="shortcut icon" href="/atlas/favicon.ico"/><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin /><link rel="preload" href="/atlas/_next/static/css/0ed7519f8f6e2b12.css" as="style"/><link rel="stylesheet" href="/atlas/_next/static/css/0ed7519f8f6e2b12.css" data-n-g=""/><link rel="preload" href="/atlas/_next/static/css/48309c2d926084e7.css" as="style"/><link rel="stylesheet" href="/atlas/_next/static/css/48309c2d926084e7.css" data-n-p=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/atlas/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script src="/atlas/_next/static/chunks/webpack-e0e074d834bfaea7.js" defer=""></script><script src="/atlas/_next/static/chunks/framework-c02f198d58d34563.js" defer=""></script><script src="/atlas/_next/static/chunks/main-88e61d9f66dc3fee.js" defer=""></script><script src="/atlas/_next/static/chunks/pages/_app-b2c4922735fa1d06.js" defer=""></script><script src="/atlas/_next/static/chunks/874-c64678d9ac232da4.js" defer=""></script><script src="/atlas/_next/static/chunks/917-19d7d6836e5ffd88.js" defer=""></script><script src="/atlas/_next/static/chunks/95-8c388448e411c861.js" defer=""></script><script src="/atlas/_next/static/chunks/pages/components/%5Bcomponent%5D-7680a099196e1d13.js" defer=""></script><script src="/atlas/_next/static/oSPjchGiWCh82QQls6uKg/_buildManifest.js" defer=""></script><script src="/atlas/_next/static/oSPjchGiWCh82QQls6uKg/_ssgManifest.js" defer=""></script><style data-href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;700&display=swap">@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuLyfMZs.woff) format('woff')}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuI6fMZs.woff) format('woff')}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuFuYMZs.woff) format('woff')}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0460-052F,U+1C80-1C88,U+20B4,U+2DE0-2DFF,U+A640-A69F,U+FE2E-FE2F}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa0ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0301,U+0400-045F,U+0490-0491,U+04B0-04B1,U+2116}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+1F00-1FFF}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0370-03FF}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0102-0103,U+0110-0111,U+0128-0129,U+0168-0169,U+01A0-01A1,U+01AF-01B0,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+1EA0-1EF9,U+20AB}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa25L7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0100-02AF,U+0304,U+0308,U+0329,U+1E00-1E9F,U+1EF2-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7W0Q5nw.woff2) format('woff2');unicode-range:U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+0304,U+0308,U+0329,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0460-052F,U+1C80-1C88,U+20B4,U+2DE0-2DFF,U+A640-A69F,U+FE2E-FE2F}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa0ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0301,U+0400-045F,U+0490-0491,U+04B0-04B1,U+2116}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+1F00-1FFF}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0370-03FF}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0102-0103,U+0110-0111,U+0128-0129,U+0168-0169,U+01A0-01A1,U+01AF-01B0,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+1EA0-1EF9,U+20AB}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa25L7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0100-02AF,U+0304,U+0308,U+0329,U+1E00-1E9F,U+1EF2-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7W0Q5nw.woff2) format('woff2');unicode-range:U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+0304,U+0308,U+0329,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0460-052F,U+1C80-1C88,U+20B4,U+2DE0-2DFF,U+A640-A69F,U+FE2E-FE2F}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa0ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0301,U+0400-045F,U+0490-0491,U+04B0-04B1,U+2116}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+1F00-1FFF}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0370-03FF}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0102-0103,U+0110-0111,U+0128-0129,U+0168-0169,U+01A0-01A1,U+01AF-01B0,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+1EA0-1EF9,U+20AB}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa25L7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0100-02AF,U+0304,U+0308,U+0329,U+1E00-1E9F,U+1EF2-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7W0Q5nw.woff2) format('woff2');unicode-range:U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+0304,U+0308,U+0329,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD}</style></head><body><div id="__next"><div><div class="Heading_Heading__EQADp"><a href="/atlas">üåê ProductPlan Atlas</a></div><div class="Documentation"><nav class="Navigation_Sidebar__ycYDm"><div class="Navigation_Sidebar__title__LoJn8">Atlas</div><ul class="Navigation_Sidebar__list__Nlfdq"><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas">Welcome</a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="https://github.com/ProductPlan/atlas-mono" target="_blank" rel="noreferrer">GitHub <i class="fa-brands fa-github"></i></a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/tokens">Tokens</a></li></ul><div class="Navigation_Sidebar__title__LoJn8">Components</div><ul class="Navigation_Sidebar__list__Nlfdq"><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/components/AtlasAlert">AtlasAlert</a></li><li class="Navigation_Sidebar__listItem__5Wet1 Navigation_Sidebar__listItemActive__3t7we"><a href="/atlas/components/AtlasBanner">AtlasBanner</a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/components/AtlasPopover">AtlasPopover</a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/components/BasicButton">BasicButton</a></li></ul></nav><div class="PageContent_Page__KQX_Q"><div class="PageContent_PageContent__Hwrl_"><h1 class="PageContent_PageHeading__H2sy3">AtlasBanner</h1><p class="PageContent_PageSubcopy__Ux2Ic">A banner which appears on load of a page and rests at the top of the page. They can be dismissible by clicking the close icon, or disappear after a certain duration.</p><div><h3>Example<!-- -->s</h3><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Banner Types</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div style="display:flex;flex-direction:column;justify-content:space-around;height:300px"><div class="AtlasBanner AtlasBanner--information"><div class="AtlasBanner__content"><i class="fas fa-circle-info"></i><div class="AtlasBanner__title">This is an information banner.</div></div></div><div class="AtlasBanner AtlasBanner--warning"><div class="AtlasBanner__content"><i class="fas fa-circle-exclamation"></i><div class="AtlasBanner__title">This is a warning banner.</div></div></div><div class="AtlasBanner AtlasBanner--destructive"><div class="AtlasBanner__content"><i class="fas fa-triangle-exclamation"></i><div class="AtlasBanner__title">This is a destructive banner.</div></div></div><div class="AtlasBanner AtlasBanner--success"><div class="AtlasBanner__content"><i class="fas fa-circle-info"></i><div class="AtlasBanner__title">This is a success banner.</div></div></div></div></div><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Banner with Button</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div><div class="AtlasBanner AtlasBanner--information AtlasBanner--button"><div class="AtlasBanner__content"><i class="fas fa-circle-info"></i><div class="AtlasBanner__title">This banner has a button.</div><atlas-basic-button align="center" label="Label" size="mini" theme="default" type="button"></atlas-basic-button></div></div></div></div><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Dismissable Banner</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div><div class="AtlasBanner AtlasBanner--warning AtlasBanner--dismissable"><div class="AtlasBanner__content"><i class="fa fa-circle-exclamation"></i><div class="AtlasBanner__title">This banner can be dismissed.</div><button aria-label="close"><i class="fas fa-xmark"></i></button></div></div></div></div><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Dismissable Banner with Button</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div><atlas-basic-button align="center" label="Show Banner" size="default" theme="default" type="button"></atlas-basic-button></div></div><h3>Props</h3><table class="Table"><thead><tr><th class="Props_TableHeading__Kvchn">Name</th><th class="Props_TableHeading__Kvchn">Description</th><th class="Props_TableHeading__Kvchn">Type</th><th class="Props_TableHeading__Kvchn">Default</th><th class="Props_TableHeading__Kvchn">Required</th></tr></thead><tbody><tr><td class="Props_TableCell__zHnLs"><strong>type</strong></td><td class="Props_TableCell__zHnLs">one of information, warning, destructive, success</td><td class="Props_TableCell__zHnLs">enum</td><td class="Props_TableCell__zHnLs">&quot;information&quot;</td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>isDismissable</strong></td><td class="Props_TableCell__zHnLs">whether or not you can close the banner</td><td class="Props_TableCell__zHnLs">bool</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>title</strong></td><td class="Props_TableCell__zHnLs">text to display for the title</td><td class="Props_TableCell__zHnLs">string</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"><i class="fa fa-check-circle"></i></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>label</strong></td><td class="Props_TableCell__zHnLs">text to display in the button</td><td class="Props_TableCell__zHnLs">string</td><td class="Props_TableCell__zHnLs">null</td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>handleClick</strong></td><td class="Props_TableCell__zHnLs">click handler</td><td class="Props_TableCell__zHnLs">func</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>handleClose</strong></td><td class="Props_TableCell__zHnLs">click handler to close banner</td><td class="Props_TableCell__zHnLs">func</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>timeout</strong></td><td class="Props_TableCell__zHnLs">how long to display the banner (in milliseconds)</td><td class="Props_TableCell__zHnLs">number</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>leadingIcon</strong></td><td class="Props_TableCell__zHnLs">icon to display to the left of banner content</td><td class="Props_TableCell__zHnLs">element</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"><i class="fa fa-check-circle"></i></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>trailingIcon</strong></td><td class="Props_TableCell__zHnLs">icon to display to the right of banner content</td><td class="Props_TableCell__zHnLs">element</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr></tbody></table></div></div></div></div></div></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{"component":"AtlasBanner"},"__N_SSG":true},"page":"/components/[component]","query":{"component":"AtlasBanner"},"buildId":"oSPjchGiWCh82QQls6uKg","assetPrefix":"/atlas","isFallback":false,"gsp":true,"scriptLoader":[]}</script></body></html>