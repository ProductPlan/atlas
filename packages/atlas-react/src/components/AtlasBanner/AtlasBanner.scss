@import "~@productplan/atlas-styles/atlas.scss";

.AtlasBanner {
  border: 4px solid var(--a-color-white-400);
  height: 60px;
  width:900px;
  background-color:var(--a-color-white-400);
  border-radius: 4px;
  box-shadow: 0px 2px 12px rgba(40, 41, 41, 0.2);

  &--information {
    color: var(--a-color-blue-700);
    background-color: var(--a-color-blue-100);
    i {
      color:  var(--a-color-blue-700)
    }
  }

  &--warning {
    color: var(--a-color-yellow-700);
    background: var(--a-color-yellow-100);
    .BasicButton {
      background-color: var(--a-color-yellow-700);
    }
    i {
      color:  var(--a-color-yellow-700)
    }
  }

  &--destructive {
    color: var(--a-color-red-700);
    background: var(--a-color-red-100);
    .BasicButton{
      background-color: var(--a-color-red-400);
    }
    i {
      color:  var(--a-color-red-400)
    }
  }

  &--success {
    color: var(--a-color-green-700);
    background: var(--a-color-green-100);
    .BasicButton {
      background-color: var(--a-color-green-500);
    }
    i {
      color: var(--a-color-green-500)
    }
  }

  &--button {
    .AtlasBanner__title {
      width: 781px;
    }
    .BasicButton {
      margin: 16px 0 16px 8px;
    }
  }

  &--dismissable {
    .AtlasBanner__title {
      width: 812px;
      margin-right: 8px;
    }
  }

   &.AtlasBanner--dismissable.AtlasBanner--button {
    .AtlasBanner__title {
      width: 745px;
    }
    .BasicButton {
      margin: 16px 16px 16px 8px;
    }
   }

   &__title {
    margin: 20px 0 20px 8px;
  }
  
  &__content {
    display: flex;
    margin: 0px 16px;
    border-radius: 4px;
    align-items: center;
    i::before {
      width: 20px;
      height: 20px;
      display: flex;
      justify-content: center;
      align-items: center;
      margin: 20px 0 20px 0;
    }
  }
  button[aria-label="close"] {
    background: transparent;
    border: none;
    padding: 0;
  }
}


