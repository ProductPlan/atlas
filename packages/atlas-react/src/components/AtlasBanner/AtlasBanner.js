import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import BasicButton from "@components/BasicButton";
import "./AtlasBanner.scss";

const bannerTypes = ["information", "warning", "destructive", "success"];
/** A banner which appears on load of a page and rests at the top of the page. They can be dismissible by clicking the close icon, or disappear after a certain duration. */
export default function AtlasBanner({
  type,
  isDismissable,
  title,
  label = null,
  handleClick,
  // timeout,
  leadingIcon,
  trailingIcon,
}) {
  const classes = classNames("AtlasBanner", { [`AtlasBanner--${type}`]: true, "AtlasBanner--dismissable": isDismissable, "AtlasBanner--button": label });

  return (
    <div className={classes}>
      <div className="AtlasBanner__content">
        <div className="AtlasBanner__required">
        {leadingIcon}
        {title && <div className="AtlasBanner__title">{title}</div>}
        </div>
        <div className="AtlasBanner__conditional">
        {label && (
          <BasicButton
            label={label}
            size="mini"
            // onClick={handleClick}
          />
        )}
        {isDismissable ? trailingIcon : null}
        </div>
      </div>
    </div>
  );
}

AtlasBanner.propTypes = {
  /** one of information, warning, destructive, success */
  type: PropTypes.oneOf(bannerTypes),
  /** whether or not you can close the banner */
  isDismissable: PropTypes.bool,
  /** text to display for the title */
  title: PropTypes.string.isRequired,
  /** text to display in the button */
  label: PropTypes.string,
  /** click handler */
  handleClick: PropTypes.func,
  /** icon to display to the left of banner content  */
  leadingIcon: PropTypes.element.isRequired,
  /** icon to display to the right of banner content  */
  trailingIcon: PropTypes.element,
};
