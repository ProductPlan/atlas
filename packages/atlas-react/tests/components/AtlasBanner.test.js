import React from 'react';
import {render, screen,fireEvent, waitFor} from '@testing-library/react';
import AtlasBanner from '@components/AtlasBanner'

describe("AtlasBanner", () => {
    it("renders banner title", () => {
        const { container } = render(
            <AtlasBanner
            title="Dogs are great."
            leadingIcon={<i className="fas fa-circle-check"></i>}
            />
        )
        const content = screen.getByText("Dogs are great.")
        expect(container).toMatchSnapshot();
        expect(content).toBeTruthy();
        })

    it("renders banner with a button", async() => {
        const handleClick= jest.fn()
        const {container} = render(
            <AtlasBanner
            label="Click Me!"
            title="Cats are great."
            leadingIcon={<i className="fas fa-circle-check"></i>}
            trailingIcon={<i className="fas fa-xmark"></i>}
            type="destructive"
            handleClick = {handleClick}
        />
        )
        expect(container).toMatchSnapshot()
        await waitFor(()=>expect(screen.getByText("Click Me!")).toBeTruthy())
        fireEvent.click(screen.getByText("Click Me!"))
        expect(handleClick).toHaveBeenCalled();
    })

    it("renders dismissable banner and checks banner is closed", () => {
        const handleClose = jest.fn();
        const {container} = render(
            <AtlasBanner
            title="Birds are great."
            leadingIcon={<i className="fas fa-circle-check"></i>}
            trailingIcon={<i className="fas fa-xmark"></i>}
            isDismissable
            handleClose = {handleClose}
        />
        )
        expect(container).toMatchSnapshot()
        const content = screen.getByText("Birds are great.")
        expect(content).toBeTruthy();
        fireEvent.click(screen.getByRole('button', {name: 'close'}))
        expect(handleClose).toHaveBeenCalled();
        expect(screen.queryByText("Birds are great.")).toBeFalsy();
    })
})
