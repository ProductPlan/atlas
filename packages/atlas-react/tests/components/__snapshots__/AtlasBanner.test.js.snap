// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtlasBanner renders banner title 1`] = `
<div>
  <div
    class="AtlasBanner AtlasBanner--information"
  >
    <div
      class="AtlasBanner__content"
    >
      <i
        class="fas fa-circle-check"
      />
      <div
        class="AtlasBanner__title"
      >
        Dogs are great.
      </div>
    </div>
  </div>
</div>
`;

exports[`AtlasBanner renders banner with a button 1`] = `
<div>
  <div
    class="AtlasBanner AtlasBanner--destructive AtlasBanner--button"
  >
    <div
      class="AtlasBanner__content"
    >
      <i
        class="fas fa-circle-check"
      />
      <div
        class="AtlasBanner__title"
      >
        Cats are great.
      </div>
      <atlas-basic-button
        align="center"
        label="Click Me!"
        size="mini"
        theme="default"
        type="button"
      />
    </div>
  </div>
</div>
`;

exports[`AtlasBanner renders dismissable banner and checks banner is closed 1`] = `
<div>
  <div
    class="AtlasBanner AtlasBanner--information AtlasBanner--dismissable"
  >
    <div
      class="AtlasBanner__content"
    >
      <i
        class="fas fa-circle-check"
      />
      <div
        class="AtlasBanner__title"
      >
        Birds are great.
      </div>
      <button
        aria-label="close"
      >
        <i
          class="fas fa-xmark"
        />
      </button>
    </div>
  </div>
</div>
`;
